service-statistics-common/src/main/java/fi/nls/oskari/control/statistics/plugins/StatisticalDatasourceFactory.java

```diff
@@ -28,15 +34,37 @@ public void setupSourceLayers(StatisticalDatasource source) {
         final DatasourceHelper helper = DatasourceHelper.getInstance();
         final DataSource dataSource = helper.getDataSource(helper.getOskariDataSourceName());
         final SqlSessionFactory factory = initializeIBatis(dataSource);
-        final SqlSession session = factory.openSession();
-        try {
+        try (final SqlSession session = factory.openSession()) {
             final List<DatasourceLayer> layerRows = session.getMapper(DatasourceLayerMapper.class).getLayersForDatasource(source.getId());
+
+            // fetch a list of permissions for the regionset layers
+            Map<Long, List<Permissions>> permissions = ServiceFactory.getPermissionsService()
```

Comment:
This is difficult to read due to current indenting and line breaks. Would be better if the maplayerIds were collected on a separate line.